### PR TITLE
if_ruby: Add rubyeval() function

### PR DESCRIPTION
**Pull request by ichizok**
Examples:

```vim
echo rubyeval('true')
" v:true

echo rubyeval('false')
" v:false

echo rubyeval('nil')
" v:null

echo rubyeval('123')
" 123

echo rubyeval('"str"')
" 'str'

echo rubyeval(':sym')
" 'sym'

echo rubyeval('[0,1,2]')
" [0,1,2]

echo rubyeval('{"key" => "val"}')
" {'key': 'val'}

" Other objects: return the result of `Object#to_s`
echo rubyeval('Vim')
" 'Vim'

echo rubyeval('Vim.class')
" 'Module'
```

This pull-req, vim converts ruby objects which cannot correspond naturally to vim objects to strings resulted from "Object#to_s".
Or, is it better to (not convert and) raise conversion error?

### CODE REVIEW

**Comment by brammool** [comment]
Ozaki Kiichi wrote:

> Examples:
> 
> ```vim
> echo rubyeval('true')
> " v:true
> 
> echo rubyeval('false')
> " v:false
> 
> echo rubyeval('nil')
> " v:null
> 
> echo rubyeval('123')
> " 123
> 
> echo rubyeval('"str"')
> " 'str'
> 
> echo rubyeval(':sym')
> " 'sym'
> 
> echo rubyeval('[0,1,2]')
> " [0,1,2]
> 
> echo rubyeval('{"key" => "val"}')
> " {'key': 'val'}
> 
> " Other objects: return the result of `Object#to_s`
> echo rubyeval('Vim')
> " 'Vim'
> 
> echo rubyeval('Vim.class')
> " 'Module'
> ```
> 
> This pull-req, vim converts ruby objects which cannot correspond naturally to vim objects to strings resulted from "Object#to_s".
> Or, is it better to (not convert and) raise conversion error?

Thanks, looks good.

In the tests you replaced RubyEval() with rubyeval().  That means the
form with ":ruby print {expr}" is no longer tested.  I think we should
test both, since the results are not exactly the same.
Can you please Keep the RubEval() lines and add the rubyeval() lines?
Perhaps with some convenient function to reduce the number of lines.  It
would take the expression and the value that RubEval() is expected to
return and a value that rubyeval() is expected to return.

-- 
hundred-and-one symptoms of being an internet addict:
102. When filling out your driver's license application, you give
     your IP address.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

File: runtime/doc/eval.txt [hunk 1]
```diff
@@ -7431,6 +7432,17 @@ round({expr})							*round()*
...
 		{only available when compiled with the |+float| feature}
 
+rubyeval({expr})					*rubyeval()*
+		Evaluate Ruby expression {expr} and return its result
+		converted to Vim data structures.
+		Numbers, floats and strings are returned as they are (strings
+		are copied though).
+		Arrays are represented as Vim |List| type.
+		Hashes are represented as Vim |Dictionary| type.
+		Othet objects are represented as strings resulted from their
```
Comment (dpelle) [inline runtime/doc/eval.txt]: Othet → Other

Comment (ichizok) [reply runtime/doc/eval.txt]: Thank you, fixed it.

**Comment by ichizok** [comment]
Updated patch: Add more tests for "ruby print" in Test_print (test_ruby).

**Pull request closed without merging**